### PR TITLE
Redundant Nulls Removal

### LINKED ISSUE #1955: Remove redundant nulls
Time Tracking: MA-181

Currently there are many nullable variables that do not need to be nullable. Likely a holdover from when Flutter added null safety. This makes variable initialization/access more complex (e.g., ? and ! operators) and less efficient due to unnecessary runtime checks for null pointers.

- [x] Pull from upstream to get the latest code
- [x] Checkout the feature branch
- [x] Compile and run the code
- [x] Develop the feature
- [x] Test locally on a device
- [x] Create Pull Request

Note:
The Model classes may be a good place to start. Anecdotally, they seem to be the source of most of the redundant nullables.
Documentation available [here](https://ucsdcollab.atlassian.net/wiki/x/nIAGTg)

### PR DESCRIPTION
**Pull request by parthshindee**
## Summary
Campus Mobile codebase contains a large number of unnecessary nullable variables and null checks. This issue removes a majority of the redundant nulls.

Resolves #1955

## Changelog
[General] [Change] - Removed redundant nulls throughout the app

## Test Plan
- [x] [PR #2055 Test Plan](https://ucsdcloud.sharepoint.com/:x:/r/sites/WorkplaceTechnologyServices-CampusMobileBuilds/_layouts/15/Doc.aspx?sourcedoc=%7B35090920-E581-45FE-9FFA-3376C4A7FB2C%7D&file=PR-2055-Test-Plan-7.29.0-QA-3299.xlsx&action=default&mobileredirect=true) PASS on iOS
- [x] [PR #2055 Test Plan](https://ucsdcloud.sharepoint.com/:x:/r/sites/WorkplaceTechnologyServices-CampusMobileBuilds/_layouts/15/Doc.aspx?sourcedoc=%7B535DED35-554C-4CA8-9139-29D4CCCB260B%7D&file=PR-2055-Test-Plan-7.29.0-QA-3300.xlsx&action=default&mobileredirect=true) PASS on Android

### CODE REVIEW

**Comment by morebytes** [comment]
Talked w/ Charles. Decided to pause this until the Dart 3 migration gets merged in

**Comment by morebytes** [comment]
Spoke with Alex. Most likely this will be paused until after the next release of Campus Mobile

**Comment by morebytes** [comment]
Finished manual merge w/ experimental. After thoroughly resolving build tools issues, there does not appear to be any regressions. Still should be monitored...

**Review by morebytes — DISMISSED** [review]
Looks good and was tested. Any issues that arise going forward will be treated as separate bug tasks

**Review by morebytes — APPROVED** [review]
Tests passed. Latest commit was a merge w/ experimental - low chance for any regressions. Looks good

File: lib/core/providers/notifications_freefood.dart [hunk 1]
```diff
@@ -36,30 +36,30 @@ class FreeFoodDataProvider extends ChangeNotifier {
...
+    _registeredEvents.remove(id);
   }
 
   void parseMessages() {
     // initializeValues();
-    List<MessageElement?> messages = _messageDataProvider.messages!;
-    messages.forEach((m) async {
-      if (m!.audience != null &&
-          m.audience!.topics != null &&
-          (m.audience!.topics!.contains("freeFood") || m.audience!.topics!.contains("testFreeFood"))) {
```
Comment (c3bryant) [inline lib/core/providers/notifications_freefood.dart:59]: @morebytes May we plan to revert or discuss for 7.31? The `testFreeFood` condition is is required to fully test Free Food notifications.

**Pull request merged**